File: lang/ruby/test/fun/vac/data_structures/LinkedQueueTest.rb
require "fun/vac/data_structures/LinkedQueue"
require "fun/vac/util/TestRunner"

def testLinkedQueue()
    size = 8192

    queue = LinkedQueue::LinkedQueue.new()

    (1 .. size).each do |i|
        queue.enqueue(i)
    end

    if queue.size() != size
        return false
    end

    (1 .. size).each do |i|
        if queue.peek() != i
            return false
        end
        queue.dequeue()
    end

    queue.isEmpty()
end

if __FILE__ == $PROGRAM_NAME
    TestRunner.parseTest(testLinkedQueue())
end

File: lang/java/test/fun/vac/data_structures/LinkedQueueTest.java
import fun.vac.data_structures.LinkedQueue;
import fun.vac.util.TestRunner;

public class LinkedQueueTest {

    public static boolean testLinkedQueue() {
        int size = 8192;

        LinkedQueue<Integer> queue = new LinkedQueue<>();

        for (int i = 1; i <= size; i++) {
            queue.enqueue(i);
        }

        if (queue.size() != size) {
            return false;
        }

        for (int i = 1; i <= size; i++) {
            if (queue.peek() != i) {
                return false;
            }
            queue.dequeue();
        }

        return queue.isEmpty();
    }

    public static void main(String[] args) {
        TestRunner.parseTest(testLinkedQueue());
    }
}
